agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/PerformanceCounterInitializer.java

```diff
@@ -109,23 +113,90 @@ private static void loadCustomJmxPerfCounters(List<Configuration.JmxMetric> jmxX
         collection.add(new JmxAttributeData(jmxElement.name, jmxElement.attribute));
       }
 
-      // Register each entry in the performance container
-      for (Map.Entry<String, Collection<JmxAttributeData>> entry : data.entrySet()) {
+      createMeterPerAttribute(data);
+
+    } catch (RuntimeException e) {
+      try (MDC.MDCCloseable ignored = CUSTOM_JMX_METRIC_ERROR.makeActive()) {
+        logger.error("Failed to register JMX performance counters: '{}'", e.toString());
+      }
+    }
```

Comment:
it looks like previously we had try/catch around each JMX metric, which could be better since then a single JMX metric problem wouldn't affect the others

Reply:
there is another try block above.